### ISSUE TITLE
Crash on app start on React Native 0.71.0 - EXC_BAD_ACCESS on realm_jsi_init

### ISSUE THREAD

**Issue by michalkos** [realm/realm-js#5259]
### How frequently does the bug occur?

Always

### Description

Realm crashes on React Native 0.71.0.

After upgrading my project to the latest RN, Realm crashes the app on start.

It even crashes on fresh new project.

### Stacktrace & log output

```shell
Thread 14 Crashed:: com.facebook.react.JavaScript
0   hermes                        	       0x10ac9551c 0x10ac8d000 + 34076
1   AwesomeProject                	       0x10722b022 facebook::jsi::RuntimeDecorator<facebook::jsi::Runtime, facebook::jsi::Runtime>::cloneObject(facebook::jsi::Runtime::PointerValue const*) + 34 (decorator.h:164)
2   AwesomeProject                	       0x107228ba2 facebook::jsi::WithRuntimeDecorator<facebook::react::(anonymous namespace)::ReentrancyCheck, facebook::jsi::Runtime, facebook::jsi::Runtime>::cloneObject(facebook::jsi::Runtime::PointerValue const*) + 50 (decorator.h:577)
3   hermes                        	       0x10aca97b0 0x10ac8d000 + 116656
4   hermes                        	       0x10acaa855 facebook::jsi::Value::asObject(facebook::jsi::Runtime&) && + 85
5   AwesomeProject                	       0x1073b1b10 realm::js::realmjsi::ObjectWrap<realm::js::ObservableClass<realm::js::realmjsi::Types> >::create_constructor(realm::js::JsiEnv) + 624
6   AwesomeProject                	       0x1073aec44 realm::js::RealmClass<realm::js::realmjsi::Types>::create_constructor(realm::js::JsiEnv) + 36
7   AwesomeProject                	       0x1073aeabe realm_jsi_init + 142
8   AwesomeProject                	       0x107344949 __24-[RealmReact setBridge:]_block_invoke + 761 (RealmReact.mm:156)
9   AwesomeProject                	       0x106f81da9 decltype(static_cast<void () block_pointer __strong&>(fp)()) std::__1::__invoke<void () block_pointer __strong&>(&&, decltype(static_cast<void () block_pointer __strong&>(fp)())&&...) + 25 (type_traits:3918)
10  AwesomeProject                	       0x106f81d5d void std::__1::__invoke_void_return_wrapper<void, true>::__call<void () block_pointer __strong&>(void () block_pointer __strong&) + 29 (invoke.h:61)
11  AwesomeProject                	       0x106f81d2d std::__1::__function::__alloc_func<void () block_pointer __strong, std::__1::allocator<std::__1::allocator>, void ()>::operator()() + 29 (function.h:178)
12  AwesomeProject                	       0x106f80979 std::__1::__function::__func<void () block_pointer __strong, std::__1::allocator<std::__1::allocator>, void ()>::operator()() + 25 (function.h:352)
13  AwesomeProject                	       0x106f83f62 std::__1::__function::__value_func<void ()>::operator()() const + 50 (function.h:505)
14  AwesomeProject                	       0x106f83f25 std::__1::function<void ()>::operator()() const + 21 (function.h:1182)
15  AwesomeProject                	       0x106f9b965 facebook::react::tryAndReturnError(std::__1::function<void ()> const&) + 21 (RCTCxxUtils.mm:74)
16  AwesomeProject                	       0x106f6a1b8 -[RCTCxxBridge _tryAndHandleError:] + 88 (RCTCxxBridge.mm:345)
17  Foundation                    	    0x7ff800c8054f __NSThreadPerformPerform + 177
18  CoreFoundation                	    0x7ff800387fe5 __CFRUNLOOP_IS_CALLING_OUT_TO_A_SOURCE0_PERFORM_FUNCTION__ + 17
19  CoreFoundation                	    0x7ff800387f24 __CFRunLoopDoSource0 + 157
20  CoreFoundation                	    0x7ff800387721 __CFRunLoopDoSources0 + 212
21  CoreFoundation                	    0x7ff800381e23 __CFRunLoopRun + 927
22  CoreFoundation                	    0x7ff8003816a7 CFRunLoopRunSpecific + 560
23  AwesomeProject                	       0x106f69fb5 +[RCTCxxBridge runRunLoop] + 949 (RCTCxxBridge.mm:336)
24  Foundation                    	    0x7ff800c8011b __NSThread__start__ + 1009
25  libsystem_pthread.dylib       	    0x7ff836176259 _pthread_start + 125
26  libsystem_pthread.dylib       	    0x7ff836171c7b thread_start + 15
```

### Can you reproduce the bug?

Always

### Reproduction Steps

Setup new project with

```
npx react-native init AwesomeProject
npm install realm
npx pod-install
```

Run the iOS project

The app crashes on start

### Version

v11.3.1

### What services are you using?

Local Database only

### Are you using encryption?

No

### Platform OS and version(s)

iOS 16.2

### Build environment

Mac OS 13.1
Intel
RN 0.71.0
iOS 16.2
Xcode 14.2

**Comment by takameyer** [comment]
@michalkos Thanks for pointing this out.  Currently Realm is tied to a specific version of React Native.  We will be rolling out a release soon, so keep an eye out.  We also have plans to break this dependency, so that React Native releases are relatively independent from Realm.

**Comment by douglasndm** [comment]
I'm having the same issue.

**Comment by RalissonMattias** [comment]
Same issue

**Comment by c100k** [comment]
Please do not comment that you have the same issue 🙏🏽. It doesn't bring anything to the problem and it pings everyone who subscribed to notifications, for nothing. You can just add a 👍🏽 to the description to show that you're concerned.

The message from @takameyer shows that the issue has been acknowledged and being resolved ;) Let's wait if we cannot help.

**Comment by takameyer** [comment]
[Release `11.3.2`](https://github.com/realm/realm-js/releases/tag/v11.3.2) has been published and is compatible with RN `0.71.0`.  Please, reach out if you hit any issues!

**Comment by kneth** [comment]
@mabdurrafey-afl Please create a new issue with all details (including stack trace).
